packages/financial-templates-lib/test/clients/OptimisticOracleEventClient.js

```diff
@@ -0,0 +1,450 @@
+      from: disputer
+    });
+    await pushPrice(correctPrice);
+
+    // Settle expired proposals and resolved disputes
+    settlementTxn1 = await optimisticOracle.settle(optimisticRequester.address, identifier, requestTime, "0x");
+    settlementTxn2 = await optimisticOracle.settle(optimisticRequester.address, identifier, requestTime + 1, "0x");
+  });
+
+  it("Return RequestPrice events", async function() {
+    await client.clearState();
+    // State is empty before update().
+    assert.deepStrictEqual([], client.getAllRequestPriceEvents());
+    await client.update();
+    assert.deepStrictEqual(client.getAllRequestPriceEvents(), [
+      {
+        transactionHash: requestTxn1.tx,
+        blockNumber: requestTxn1.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: "0x",
+        timestamp: requestTime.toString(),
+        currency: collateral.address,
+        reward: "0",
+        finalFee
+      },
+      {
+        transactionHash: requestTxn2.tx,
+        blockNumber: requestTxn2.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: "0x",
+        timestamp: (requestTime + 1).toString(),
+        currency: collateral.address,
+        reward: "0",
+        finalFee
+      }
+    ]);
+
+    // Correctly adds only new events after last query
+    const newTxn = await optimisticRequester.requestPrice(
+      identifier,
+      requestTime,
+      // Note: we're using collateral address as ancillary data here to test with more entropy.
+      collateral.address,
+      collateral.address,
+      0
+    );
+    await client.clearState();
+    await client.update();
+    assert.deepStrictEqual(client.getAllRequestPriceEvents(), [
+      {
+        transactionHash: newTxn.tx,
+        blockNumber: newTxn.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        // Note: Convert contract address to lowercase to adjust for how Solidity casts addresses to bytes.
+        // This is important because `requestPrice` expects `ancillaryData` to be of type bytes,
+        ancillaryData: collateral.address.toLowerCase(),
+        timestamp: requestTime.toString(),
+        currency: collateral.address,
+        reward: "0",
+        finalFee
+      }
+    ]);
+  });
+  it("Return ProposePrice events", async function() {
+    await client.clearState();
+    // State is empty before update().
+    assert.deepStrictEqual([], client.getAllProposePriceEvents());
+    await client.update();
+    assert.deepStrictEqual(client.getAllProposePriceEvents(), [
+      {
+        transactionHash: proposalTxn1.tx,
+        blockNumber: proposalTxn1.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        proposer,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: "0x",
```

Comment:
nit: do you want to create a `defaultAncillaryData` var?

Reply:
Dont and removed other hardcoded test constants in `monitors/OptimisticOracleContractMonitor`